spec/system/candidate_interface/preferences/candidate_adds_preferences_spec.rb

```diff
@@ -277,4 +337,30 @@ def then_i_am_redirected_to_opt_in_page
   def when_i_click_change_on_the_last_location
     all('a', text: 'Change').last.click
   end
+
+  def and_i_complete_the_flow_for_adding_a_choice
+    choose 'Yes, I know where I want to apply'
+    click_link_or_button('Continue')
+
+    select 'Gorse SCITT (1N1)'
+    click_link_or_button('Continue')
+
+    choose 'Mathematics (SEND) (C998)'
+    click_link_or_button('Continue')
+
+    click_link_or_button('Review application')
+    click_link_or_button('Confirm and submit application')
+  end
+
+  def and_i_click_the_relevant_change_link
+    all(:link, 'Change')[1].click
+  end
+
+  def when_i_click_on_the_dynamic_location_change_link
+    all(:link, 'Change')[2].click
```

Comment:
Is there not visually hidden text in the change link? If there isn't we will need to add it so that the change links are all uniq.